packages/livechat/src/lib/hooks.js

```diff
@@ -60,13 +60,17 @@ const api = {
 		});
 	},
 
-	setDepartment(value) {
-		const { config: { departments = [] } } = store.state;
+	async setDepartment(value) {
+		const { config: { departments = [] }, department: existingDepartment } = store.state;
 
 		const dept = departments.find((dep) => dep._id === value || dep.name === value);
 		const department = (dept && dept._id) || '';
 
 		updateIframeGuestData({ department });
+
+		if (department !== existingDepartment) {
+			await loadConfig();
```

Comment:
won't this cause the same issue we had some time ago where the room got "discarded" on the UI so the visitor had to open a new one on livechat?

Reply:
Yups, you're right, that would indeed be a problem. I think we'll need to reload the messages too after config changes

Reply:
Tested the new change and it works. Here's the recording of my test -> https://www.loom.com/share/fab58ba6e1fe48238784d15c5c0b2bc3 Let me know if I miss anything here